tms/src/main/java/com/netcracker/edu/tms/web/Controller.java

```diff
@@ -0,0 +1,90 @@
+package com.netcracker.edu.tms.web;
+
+import com.netcracker.edu.tms.model.Priority;
+import com.netcracker.edu.tms.model.Status;
+import com.netcracker.edu.tms.model.Task;
+import com.netcracker.edu.tms.service.TaskService;
+import org.apache.velocity.exception.ResourceNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.math.BigInteger;
+import java.text.ParseException;
+import java.util.List;
+
+@RequestMapping(value = "/tasks")
+@RestController
+public class Controller {
+    @Autowired
+    private TaskService taskService;
+
+    @GetMapping("/getTaskById")
+    public ResponseEntity<Task> getTaskById(@RequestParam(value = "taskId", required = true) BigInteger taskId) {
+        return ResponseEntity.ok(taskService.getTaskById(taskId));
+    }
+
+    @GetMapping("/listOfTasksByName")
+    public @ResponseBody
+    List<Task> listOfTasksByName(@RequestParam("taskName") String taskName) {
+        return taskService.listOfTasksByName(taskName);
+    }
+
+    @PostMapping()
+    boolean addTask(@RequestBody Task task) {
+        return taskService.addTask(task);
+    }
+
+    @PutMapping
+    boolean updateTask(@RequestBody Task task) {
+        return taskService.updateTask(task);
+    }
+
+    @DeleteMapping
+    ResponseEntity deleteTask(@RequestBody Task task) {
+        try {
+            return new ResponseEntity(taskService.deleteTask(task), HttpStatus.OK);
+        } catch (ResourceNotFoundException ex) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "task " + task.getTaskId() + " not found", ex);
+        }
+
+    }
+
+    @GetMapping("/listOfTasksByReporter")
```

Comment:
it is better to leave @GetMapping only

in this case URI looks like
/tasks?reporterid={reporterId}

it means return tasks for which reporterId =  {reporterId}

Reply:
Current version is
/tasks/listOfTasksByReporter?reporterid={reporterId}

i think listOfTasksByReporter is redudant part of path

Reply:
the same for methods below